src/helpers/enrich-dataset.ts

```diff
@@ -0,0 +1,186 @@
+        } as UserSession) + '?f=json',
+        language: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.dataLang.languageCode.@_value') || localeToLang(datasetAttr.culture) || '',
+        keyword: getDatasetKeyword(datasetAttr),
+        issuedDateTime: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.idCitation.date.pubDate') || timestampToIsoDate(datasetAttr.created),
+        orgTitle,
+        provenance: _.get(datasetAttr, 'metadata.metadata.dataIdInfo.idCredit', ''),
+        hubLandingPage: concatUrlAndPath(siteUrl, relative.slice(1)),
+        downloadLink: concatUrlAndPath(siteUrl, `datasets/${identifier}`),
+        agoLandingPage: getAgoLandingPageUrl(dataset.id, portalUrl),
+        isLayer: isLayer(datasetAttr),
+        license: getDatasetLicense(datasetAttr)
+    };
+
+    const downloadLinkFor: (type: string) => string = getDownloadLinkFn(additionalFields.downloadLink, datasetAttr);
+
+    if (isProxiedCSV(datasetAttr)) {
+        additionalFields.accessUrlCSV = downloadLinkFor('csv');
+    }
+
+    if (isLayer(datasetAttr)) {
+        additionalFields.durableUrlGeoJSON = generateDurableDownloadUrl(dataset.id, siteUrl, 'geojson');
+        additionalFields.durableUrlCSV = generateDurableDownloadUrl(dataset.id, siteUrl, 'csv');
+        if (_.has(datasetAttr, 'layer.geometryType')) {
+            additionalFields.durableUrlKML = generateDurableDownloadUrl(dataset.id, siteUrl, 'kml');
+            additionalFields.durableUrlShapeFile = generateDurableDownloadUrl(dataset.id, siteUrl, 'shapefile');
+        }
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WFS_SERVER)) {
+        additionalFields.accessUrlWFS = ogcUrl(datasetAttr.url, 'WFS');
+    }
+
+    if (datasetAttr.supportedExtensions?.includes(WMS_SERVER)) {
+        additionalFields.accessUrlWMS = ogcUrl(datasetAttr.url, 'WMS');
+    }
+
+    dataset.properties = {
+        ...dataset.properties,
+        ...additionalFields
+    };
+
+    return dataset;
+};
+
+function generateDurableDownloadUrl(datasetId: string, siteUrl: string, fileType: FileType) {
+    const { itemId, layerId } = parseDatasetId(datasetId);
+    return `https://${siteUrl}/api/download/v1/items/${itemId}/${fileType}?layers=${layerId}`;
+}
+
+function getDatasetKeyword(dataset: HubDataset): string[] {
+    const metaKeyword = _.get(dataset, 'metadata.metadata.dataIdInfo.searchKeys.keyword');
+
+    if (metaKeyword) {
+        return metaKeyword;
+    }
+
+    const { tags, type, typeKeywords } = dataset;
+    const hasNoTags = !tags || tags.length === 0 || !tags[0]; // if tags is undefined, the tags array is empty, or tags is an empty string
+
+    if (isPage({ type, typeKeywords } as IItem) && hasNoTags) {
+        return ['ArcGIS Hub page'];
+    }
+
+    return tags;
+}
+
+function localeToLang(locale: string) {
+    return locale ? alpha2ToAlpha3Langs[locale.split('-')[0]] : '';
+}
+
+function isLayer(hubDataset: HubDataset): boolean {
+    return /_/.test(hubDataset.id);
```

Comment:
This will always return false since `hubDataset.id` will always be the `item.id` when coming back from the OGC API

Reply:
Same as above.